rust/connlib/tunnel/src/peer.rs

```diff
@@ -225,9 +226,7 @@ where
             TunnResult::WriteToNetwork(packet) => {
                 let mut packets = VecDeque::from([Bytes::copy_from_slice(packet)]);
 
-                // Boringtun requires us to call `decapsulate` repeatedly if it returned `WriteToNetwork`.
-                // However, for the repeated calls, we only need a buffer of at most 148 bytes which we can easily allocate on the stack.
-                let mut buf = [0u8; 148];
+                let mut buf = [0u8; MAX_UDP_SIZE];
```

Comment:
That is gonna be some big stacks repeatedly. I am surprised that we need a big buffer here. I thought I had traced all codepaths codepaths from `decapsulate`. I'll have a lool again if you don't mind.